Add information about after clause timeout limit

michalmuskala opened the pull request:
Source: http://erlang.org/doc/reference_manual/expressions.html#id80928

eksperimental commented on lib/elixir/lib/kernel/special_forms.ex:
```diff
@@ -1908,6 +1908,10 @@ defmodule Kernel.SpecialForms do
     * `0` - if there is no matching message in the mailbox, the timeout
       will occur immediately
+      
+  The timeout value must be a positive integer smaller than 0xffffffff or
```
The timeout value must be a **non-negative integer**

eksperimental replied:
We need backticks around the numbers `0xffffffff`

lexmag replied:
Maybe the following will read better:
```
smaller than `4_294_967_295` (`0xFFFFFFFF` in hex notation)
```
_Note, I used uppercase in hex notation: https://github.com/lexmag/elixir-style-guide#hex-literals._

eksperimental commented on lib/elixir/lib/kernel/special_forms.ex:
```diff
@@ -1908,6 +1908,10 @@ defmodule Kernel.SpecialForms do
       will occur immediately
+      
+  The timeout value must be a positive integer smaller than 0xffffffff or
+  4294967295 - it should be possible to represent the timeout value as an
```
I would separate that huge number with underscores as `4_294_967_295`

josevalim commented:
Agreed with @lexmag.
-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

michalmuskala commented:
Updated

eksperimental commented:
sorry, but it still says that it must be a positive number, while `0` is still a valid timeout.

michalmuskala commented:
You're right, I forgot about that change.

But actually thinking about it, we list "special values" just above - it includes zero. So I would say, zero is already handled in the definition. Maybe we should introduce it with "The regular timeout value" to differentiate from the "special values".

eksperimental commented:
:+1 on differentiating special from regular values.

michalmuskala commented:
I reworked this a bit more now.

The docs were incorrect since they said the timeout can be a variable, when in fact, any expression evaluating to the allowed values will do. I also changed the structure slightly so that we could merge the special value with the regular value definitions.

eksperimental commented on lib/elixir/lib/kernel/special_forms.ex:
```diff
@@ -1900,15 +1900,19 @@ defmodule Kernel.SpecialForms do
     * `0` - if there is no matching message in the mailbox, the timeout
       will occur immediately
 
+    * positive integer smaller than `4_294_967_295` (`0xFFFFFFFF `
```
extra space after 0xFFFFFFFF

lexmag commented on lib/elixir/lib/kernel/special_forms.ex:
```diff
@@ -1900,15 +1900,19 @@ defmodule Kernel.SpecialForms do
   The `after` clause can be specified even if there are no match clauses.
-  The timeout value given to `after` can be a variable; two special
-  values are allowed:
+  The timeout value given to `after` can be any expression evaluating to
```
So much better wording. 👏

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
